src/utils.rs

```diff
@@ -50,6 +50,58 @@ fn strip_verbatim_from_prefix(prefix: &PrefixComponent<'_>) -> Option<PathBuf> {
     Some(ret)
 }
 
+#[derive(Debug)]
+struct RemoveError {
+    kind: std::io::ErrorKind,
+    path: String,
+}
+
+impl std::error::Error for RemoveError {}
+
+impl std::fmt::Display for RemoveError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!(
+            "failed to remove '{}' : {:?}",
+            self.path, self.kind
+        ))
+    }
+}
+
+pub(crate) fn improve_remove_error(error: std::io::Error, path: &Path) -> std::io::Error {
```

Comment:
I feel like it'd be better to create custom `remove_file` and `remove_dir_all` functions than using `.map_err` all over the codebase.